docs/en/latest/architecture-design/apisix.md

```diff
@@ -21,25 +25,160 @@ title: APISIX
 #
 -->
 
-## Apache APISIX : Software Architecture
+APISIX is built on top of Nginx and [ngx_lua](https://github.com/openresty/lua-nginx-module) leveraging the power offered by LuaJIT. See [Why Apache APISIX chose Nginx and Lua to build API Gateway?](https://apisix.apache.org/blog/2021/08/25/why-apache-apisix-chose-nginx-and-lua/).
+
+![flow-software-architecture](https://raw.githubusercontent.com/apache/apisix/master/docs/assets/images/flow-software-architecture.png)
+
+APISIX has two main parts:
+
+1. APISIX core, Lua plugin, multi-language Plugin runtime, and the WASM plugin runtime.
+2. Built-in Plugins that adds features for observability, security, traffic control, etc.
+
+The APISIX core handles the important functions like matching Routes, load balancing, service discovery, configuration management, and provides a management API. It also includes APISIX Plugin runtime supporting Lua and multilingual Plugins (Go, Java , Python, JavaScript, etc) including the experimental WASM Plugin runtime.
+
+APISIX also has a set of [built-in Plugins](https://apisix.apache.org/docs/apisix/plugins/batch-requests) that adds features like authentication, security, observability, etc. They are written in Lua.
+
+## Deployment modes
```

Comment:
Hi @navendu-pottekkat, may I know the reason to merge the Deployment part with the architecture part? As I think, the Deployment section is a standalone page to document. What's your suggestion? @nfrankel

Reply:
It could be an extension of the APISIX architecture or how it differs for different production use cases. Since this feature will be official only in v3, we will have to change the installation and getting started pages to reflect it. Maybe this can be moved to some place else.

Reply:
Agreed with @juzhiyuan. Though architecture and deployment may be related, they are different concerns.

As a first-time visitor, I'm much more interested in the architecture (_e.g._, plugin-based) than in how to deploy. Only when my interest is piqued will I review the different deployment options.

Reply:
>Or should we move it to the installation docs.

IMHO, it's separate from installation

>Since this feature is not enabled by default

I don't think we should reason about it this way. The are multiple modes and one of them is default.

>Maybe refactor installation docs in v3 release mentioning deployment modes by default?

We should mention the default very prominently in the install section and then reference deployments modes as a link.

Reply:
Okay. That sounds good. I will make the update and ping for review.

Reply:
Deployment modes is a separate page and it is referenced in the installation docs.